nvtabular/ops/categorify.py

```diff
@@ -457,24 +460,43 @@ def _get_aggregation_type(col):
 def _write_gb_stats(dfs, base_path, col_group, on_host, concat_groups, name_sep):
     if concat_groups and len(col_group) > 1:
         col_group = [_make_name(*col_group, sep=name_sep)]
-    ignore_index = True
-    df = _concat(dfs, ignore_index)
-    if on_host:
-        df.reset_index(drop=True, inplace=True)
-        df = cudf.from_pandas(df)
     if isinstance(col_group, str):
         col_group = [col_group]
+
     rel_path = "cat_stats.%s.parquet" % (_make_name(*col_group, sep=name_sep))
     path = os.path.join(base_path, rel_path)
-    if len(df):
-        df = df.sort_values(col_group, na_position="first")
-        df.to_parquet(path, write_index=False, compression=None)
-    else:
-        df_null = cudf.DataFrame({c: [None] for c in col_group})
-        for c in col_group:
-            df_null[c] = df_null[c].astype(df[c].dtype)
-        df_null.to_parquet(path, write_index=False, compression=None)
-    del df
+    pwriter = None
+    pa_schema = None
+    if not on_host:
+        pwriter = ParquetWriter(path, compression=None)
+
+    # Loop over dfs and append to file
+    # TODO: For high-cardinality columns, should support
+    #       Dask-based to_parquet call here (but would need to
+    #       support directory reading within dependent ops)
+    n_writes = 0
+    for df in dfs:
+        if len(df):
+            if on_host:
+                # Use pyarrow
+                pa_table = pa.Table.from_pandas(df, schema=pa_schema, preserve_index=False)
+                if pwriter is None:
+                    pa_schema = pa_table.schema
+                    pwriter = pq.ParquetWriter(path, pa_schema, compression=None)
+                pwriter.write_table(pa_table)
+            else:
+                # Use CuDF
+                df.reset_index(drop=True, inplace=True)
+                pwriter.write_table(df)
+            n_writes += 1
+
+    # Close writer and return path
+    pwriter.close()
```

Comment:
Maybe we should move the ```pwriter.close()``` to after the ```if n_writes == 0``` check?  When there is no data written out, and on_host = True - I believe this line will throw a ```AttributeError: 'NoneType' object has no attribute 'close'``` exception